presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java

```diff
@@ -306,6 +323,35 @@ private static ConnectorPageSource createParquetPageSource(
         }
     }
 
+    public static Optional<org.apache.parquet.schema.Type> getColumnType(
+            Map<Integer, org.apache.parquet.schema.Type> parquetIdToField,
+            MessageType messageType,
+            IcebergColumnHandle column)
+    {
+        if (isPushedDownSubfield(column)) {
+            Subfield pushedDownSubfield = getPushedDownSubfield(column);
+            return getSubfieldType(messageType, pushedDownSubfield.getRootName(), nestedColumnPath(pushedDownSubfield));
+        }
+
+        if (parquetIdToField.isEmpty()) {
+            // This is a migrated table
+            return Optional.ofNullable(getParquetTypeByName(column.getName(), messageType));
+        }
+        return Optional.ofNullable(parquetIdToField.get(column.getId()));
+    }
+
+    private static HiveColumnHandle.ColumnType getHiveColumnHandleColumnType(IcebergColumnHandle.ColumnType columnType)
+    {
+        switch (columnType) {
+            case REGULAR:
+                return REGULAR;
+            case SYNTHESIZED:
+                return SYNTHESIZED;
+        }
+
+        throw new PrestoException(GENERIC_INTERNAL_ERROR, "Unknown ColumnType: " + columnType);
+    }
```

Comment:
This method looks a bit unclear to me. Why would you like to maintain a method to return REGULAR when REGULAR and SYNTHESIZED when SYNTHESIZED?

Reply:
This piece of code actually converts the `ColumnType` of `IcebergColumnHandle` to the `ColumnType` of `HiveColumnHandle`.